### PR TITLE
cNFT asserts

### PR DESCRIPTION
**Pull request by 0x5CE**
### init
- [x] Panic if `state` already exists
### mint
- [x] Panic if `deposit_attached` is less than `mint_price`
- [x] Panic if `total_supply` greater than or equal to `max_copies`
- [x] Panic if `tokenId` already exists

### nft_transfer
- [x] Panic if `predecessor_account_id ` is not `owner_id`
- [x] Panic if `bidder_id` is same as `owner_id`
- [ ] Panic if `token` is burned.

### set_bid
- [x] Panic if `amount` is zero
- [x] Panic if `amount` is not same a `attached_deposit`
- [x] Panic if `token` doesn't exist
- [ ] Panic if `token` is burned.
### accept_bid
- [x] Panic if `predecessor_account_id ` is not `owner_id`
- [ ] Panic if `token` is burned.

---

### CODE REVIEW

File: as/cNFT/assembly/core.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { TokenId } from './types'
 import { Token, persistent_tokens } from './models/persistent_tokens'
 import { persistent_tokens_metadata } from './models/persistent_tokens_metadata'
 import { NftEventLogData, NftTransferLog } from './models/log'
-import { logging } from 'near-sdk-as'
+import { logging, context } from 'near-sdk-as'
+import { assert_one_yocto } from '../../NFT/assembly/asserts'
```
Comment (achrafismyname) [inline as/cNFT/assembly/core.ts:7]: wrong import here

**Review by achrafismyname — APPROVED** [review]
Seems good :)

**Review by devey — CHANGES_REQUESTED** [review]
Great, it's coming together! :) I just left a few minor comments, then we can merge it!

**Review by AshenI99 — APPROVED** [review]
Seems good for me other than the things that Yassine mentioned. 👍

**Review by devey — APPROVED** [review]
Good to merge

**Pull request merged**